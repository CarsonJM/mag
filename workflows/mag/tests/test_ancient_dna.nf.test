nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_ancient_dna"

    test("Parameters: ancient_dna = true") {

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                ).match()}
            )
        }
    }
}
