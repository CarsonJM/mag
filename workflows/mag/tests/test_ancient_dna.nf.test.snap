{
    "Parameters: ancient_dna = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut.contigs.fa:md5,3055734183fc34cba4b50afbda4f34cf"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_scaffolds.fasta:md5,acb79980009afc87e0e81293a84fe54f"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut_sample2.contigs.fa:md5,093b87fe274ddb7f7dbf5d5e574495ad"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_sample2_scaffolds.fasta:md5,98a1c0bf47c301bbd838b32a54b7537b"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "MEGAHIT-MetaBAT2Refined-test_minigut.1.fa:md5,c010aca18e5dfa00f17b24ce507988f3",
                            "MEGAHIT-MetaBAT2Refined-test_minigut.2.fa:md5,2cc38f264e02be8fab0867b1dfe9f9bd"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut.001.fa:md5,1e01c5c341311b94df0f5b0b97668cde",
                            "SPAdes-MaxBin2Refined-test_minigut.002.fa:md5,ca74ae68469c1e354a00d7a1bf0416d0"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut_sample2.001.fa:md5,7c8439fb9c3b72ba4d89b9b61b538a66"
                        ]
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "MEGAHIT-DASToolUnbinned-test_minigut.fa:md5,4b022bfb579d9873385aadc9eaf549d6"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "SPAdes-DASToolUnbinned-test_minigut_sample2.fa:md5,dd179320159df68d6bf630d6b46079b5"
                    ]
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,3550550b622b6d5f6803ffed9742e08b",
                    "versions.yml:md5,368d757df0f57b49bce09965641cc71a",
                    "versions.yml:md5,47df50b983a036f55cb936247d920744",
                    "versions.yml:md5,586dc48e8e2cc8106cb0ee3b70ce0fef",
                    "versions.yml:md5,6d41fb977d31ce96d7ddbb4ea2dbe38b",
                    "versions.yml:md5,7017c8959eb098440346ca9a2310c460",
                    "versions.yml:md5,7ccffaba4e75be98b8faedb54ccea16f",
                    "versions.yml:md5,8187d1683f97b64a582620caba2b175a",
                    "versions.yml:md5,82c86a3130605a0de153c903b2cfc6d2",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,aee39cb21305f6d3fe88bf9863529706",
                    "versions.yml:md5,b4ee8f08c1efaac6dee44f113c7a2e0b",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,c384e0cacaa344670dcde728d78a524a",
                    "versions.yml:md5,c777f00c6a48f3c0490d09b1aa8d81a6",
                    "versions.yml:md5,cc4a4972327dec07faf1f54ead653715",
                    "versions.yml:md5,d8d2321bec0c70de2f9750f258fc890d",
                    "versions.yml:md5,dc29f891cb297131cf940ba5580a6832",
                    "versions.yml:md5,df654dd4fdb4039c2d8cf0b6b49f4b55",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0",
                    "versions.yml:md5,e9f8fe610b5abe68b28d77efc951e9e6"
                ],
                "assemblies": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut.contigs.fa:md5,3055734183fc34cba4b50afbda4f34cf"
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_scaffolds.fasta:md5,acb79980009afc87e0e81293a84fe54f"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-test_minigut_sample2.contigs.fa:md5,093b87fe274ddb7f7dbf5d5e574495ad"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes"
                        },
                        "SPAdes-test_minigut_sample2_scaffolds.fasta:md5,98a1c0bf47c301bbd838b32a54b7537b"
                    ]
                ],
                "bin_summary": [
                    
                ],
                "genomad": [
                    
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "MEGAHIT-MetaBAT2Refined-test_minigut.1.fa:md5,c010aca18e5dfa00f17b24ce507988f3",
                            "MEGAHIT-MetaBAT2Refined-test_minigut.2.fa:md5,2cc38f264e02be8fab0867b1dfe9f9bd"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut.001.fa:md5,1e01c5c341311b94df0f5b0b97668cde",
                            "SPAdes-MaxBin2Refined-test_minigut.002.fa:md5,ca74ae68469c1e354a00d7a1bf0416d0"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "binner": "DASTool",
                            "refinement": "dastool_refined",
                            "domain": "unclassified"
                        },
                        [
                            "SPAdes-MaxBin2Refined-test_minigut_sample2.001.fa:md5,7c8439fb9c3b72ba4d89b9b61b538a66"
                        ]
                    ]
                ],
                "refined_unbins": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "MEGAHIT-DASToolUnbinned-test_minigut.fa:md5,4b022bfb579d9873385aadc9eaf549d6"
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false,
                            "assembler": "SPAdes",
                            "refinement": "dastool_refined_unbinned",
                            "binner": "DASTool",
                            "domain": "unclassified"
                        },
                        "SPAdes-DASToolUnbinned-test_minigut_sample2.fa:md5,dd179320159df68d6bf630d6b46079b5"
                    ]
                ],
                "short_reads": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_R2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,0f67b0839fcef0334e0dcaf5568be9e0",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,2d4f5bd36e073ac683075e4bc52884be",
                    "versions.yml:md5,3550550b622b6d5f6803ffed9742e08b",
                    "versions.yml:md5,368d757df0f57b49bce09965641cc71a",
                    "versions.yml:md5,47df50b983a036f55cb936247d920744",
                    "versions.yml:md5,586dc48e8e2cc8106cb0ee3b70ce0fef",
                    "versions.yml:md5,6d41fb977d31ce96d7ddbb4ea2dbe38b",
                    "versions.yml:md5,7017c8959eb098440346ca9a2310c460",
                    "versions.yml:md5,7ccffaba4e75be98b8faedb54ccea16f",
                    "versions.yml:md5,8187d1683f97b64a582620caba2b175a",
                    "versions.yml:md5,82c86a3130605a0de153c903b2cfc6d2",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,aee39cb21305f6d3fe88bf9863529706",
                    "versions.yml:md5,b4ee8f08c1efaac6dee44f113c7a2e0b",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,bb7110425028f7e7abcbf17152405079",
                    "versions.yml:md5,c384e0cacaa344670dcde728d78a524a",
                    "versions.yml:md5,c777f00c6a48f3c0490d09b1aa8d81a6",
                    "versions.yml:md5,cc4a4972327dec07faf1f54ead653715",
                    "versions.yml:md5,d8d2321bec0c70de2f9750f258fc890d",
                    "versions.yml:md5,dc29f891cb297131cf940ba5580a6832",
                    "versions.yml:md5,df654dd4fdb4039c2d8cf0b6b49f4b55",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0",
                    "versions.yml:md5,e9f8fe610b5abe68b28d77efc951e9e6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-08T11:19:54.901540968"
    }
}